src/main/scala/inox/tip/TipDebugger.scala

```diff
@@ -47,9 +47,11 @@ trait TipDebugger extends Solver {
     }
   }
 
-  abstract override def assertCnstr(expr: Expr): Unit = {
+  abstract override def assertCnstr(expr: Expr): Unit = try {
     debugOut.foreach { o => o.printScript(encoder.encode(expr)) }
     super.assertCnstr(expr)
+  } catch {
+    case te: program.symbols.TypeErrorException => ctx.reporter.fatalError(te.pos, te.getMessage)
```

Comment:
Not sure it's the best place to add this catch, but it seemed to be the closest to the frontier between Inox and Stainless and at the same time required the less changes. This at least makes stainless reports errors more nicely. I noticed however that not all `Solver`s available through the `SolverFactory` were mixed in with `TipDebugger` so this is probably not the most robust solution. @samarion, maybe you have a better idea in mind?

---

Before:
```
[  Info  ]  - Now considering 'postcondition' VC for fun @?:?...
<stack trace of uncaught exception from Main>
```

Now:
```
[  Info  ]  - Now considering 'postcondition' VC for fun @?:?...
[ Fatal  ] Type error: <a proper error message>
<graceful exit>
```